src/libraries/System.Security.Cryptography.X509Certificates/tests/DynamicChainTests.cs

```diff
@@ -201,6 +201,97 @@ void CheckChain()
             }
         }
 
+        [Fact]
+        public static void TestBasicConstraintsViolation()
+        {
+            byte[] serialNumber = new byte[9];
+            RandomNumberGenerator.Fill(serialNumber);
+            serialNumber[0] &= 0b01111111;
```

Comment:
99% sure CertificateRequest handles this already, by inserting an 0x00 if the high bit was set.